### PR TITLE
.

### PR DESCRIPTION
**Pull request by JustZerooo**
1

### CODE REVIEW

File: sql/updates/world/master/2020_07_07_00_world.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `spell_script_names` WHERE `spell_id` = 109132;
+INSERT INTO `world`.`spell_script_names` (`spell_id`, `ScriptName`) VALUES ('109132', 'spell_monk_roll');
```
Comment (Jinnaix) [inline sql/updates/world/master/2020_07_07_00_world.sql:2]: This query would never apply to a database named anything else than "world" - please adjust this by removing it.

**Review by Jinnaix — CHANGES_REQUESTED** [review]

File: sql/updates/world/master/2020_07_07_00_world.sql (hunk 1 above)
Comment (DerEnderman) [reply sql/updates/world/master/2020_07_07_00_world.sql:2]: No, because it is in world folder and is correct. Sry i dont change

**Comment by Ovahlord** [comment]
You should update the spell script to the modern scripting model to keep the standard for that file going.

**Comment by DerEnderman** [comment]
> 
> 
> You should update the spell script to the modern scripting model to keep the standard for that file going.

Working fine no need to update and not interested. it works and that is all we need if you dont like then dont use

**Comment by MaxtorCoder** [comment]
> 
> 
> > You should update the spell script to the modern scripting model to keep the standard for that file going.
> 
> Working fine no need to update and not interested. it works and that is all we need if you dont like then dont use

It's about that people follow the simple coding standards, if you cannot follow that then why bother contributing in the first place. :)

File: src/server/scripts/Spells/spell_monk.cpp [hunk 2]
```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
...
+  public:
+    spell_monk_roll() : SpellScriptLoader("spell_monk_roll") { }
+
+    class spell_monk_roll_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_monk_roll_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_MONK_ROLL });
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:179]: You don't need to validate existence of the spell that is being scripted (checked externally in ObjectMgr)
You need to validate that other spells that you use in this script.

File: src/server/scripts/Spells/spell_monk.cpp [hunk 3]
```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
...
+        {
+            Aura* aur = GetCaster()->AddAura(SPELL_MONK_ROLL_TRIGGER, GetCaster());
+            if (!aur)
+                return;
+
+            AuraApplication* app = aur->GetApplicationOfTarget(GetCaster()->GetGUID());
+            if (!app)
+                return;
+
+            app->ClientUpdate();
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:192]: Why is this part needed at all?

File: src/server/scripts/Spells/spell_monk.cpp [hunk 4]
```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
...
+        }
+
+        void HandleAfterCast()
+        {
+            Unit* caster = GetCaster();
+            if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:198]: caster in SpellScript is never nullptr
nothing in this script requires caster to also be a player

File: src/server/scripts/Spells/spell_monk.cpp [hunk 5]
```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
     }
 };
 
+// Roll - 109132 or Roll (3 charges) - 121827
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:167]: Why does the comment mention 121827 when that spell is not given the script in database?

File: src/server/scripts/Spells/spell_monk.cpp [hunk 6]
```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
...
+        void HandleAfterCast()
+        {
+            Unit* caster = GetCaster();
+            if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
+                return;
+
+            caster->CastSpell(caster, SPELL_MONK_ROLL_TRIGGER, true);
+
+            if (caster->HasAura(SPELL_MONK_ITEM_PVP_GLOVES_BONUS))
+                caster->RemoveAurasByType(SPELL_AURA_MOD_DECREASE_SPEED);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:204]: Shouldn't this be done by casting 125127 instead of manually removing aura type?

File: src/server/scripts/Spells/spell_monk.cpp [hunk 7]
```diff
@@ -39,6 +39,11 @@ enum MonkSpells
     SPELL_MONK_SOOTHING_MIST                            = 115175,
     SPELL_MONK_STANCE_OF_THE_SPIRITED_CRANE             = 154436,
     SPELL_MONK_SURGING_MIST_HEAL                        = 116995,
+    SPELL_MONK_ROLL                                     = 109132,
+    SPELL_MONK_ROLL_TRIGGER                             = 107427,
+    SPELL_MONK_ITEM_PVP_GLOVES_BONUS                    = 124489,
+    SPELL_MONK_DISABLE                                  = 116095,
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_monk.cpp:45]: What do disable spells have to do with roll?

File: src/server/scripts/Spells/spell_monk.cpp (hunk 6 above)
Comment (Shauren) [reply src/server/scripts/Spells/spell_monk.cpp:204]: Actually, no item has this effect attached to it, why is it here at all

**Review by Shauren — CHANGES_REQUESTED** [review]
